### ISSUE TITLE
Property returns incorrect callable type

### ISSUE THREAD

**Issue by ermishechkin** [microsoft/pyright#1150]
**Describe the bug**
Property type is incorrect if getter returns callable (first argument is dropped):

**Screenshots or Code**
```python
class Klass:
    def __init__(self, converter: Callable[[str, int], int]) -> None:
        self.converter = converter

    @property
    def converter_prop(self) -> Callable[[str, int], int]:
        return self.converter

def str_to_int(arg: str, base: int) -> int:
    return int(arg, base=base)

obj = Klass(str_to_int)
val1: int = obj.converter('123', 10)  # OK
val2: int = obj.converter_prop('123', 10)  # error: Expected 1 positional argument
# error: Argument of type "Literal['123']" cannot be assigned to parameter of type "int"
#        "Literal['123']" is incompatible with "int" (reportGeneralTypeIssues)

reveal_type(obj.converter)       # Type of "obj.converter" is "(p0: str, p1: int) -> int"
reveal_type(obj.converter_prop)  # Type of "obj.converter_prop" is "(p1: int) -> int"
```

**VS Code extension or command-line**
pyright 1.1.83
Python 3.8.0

**Comment by erictraut** [comment]
Thanks for the bug report. I've confirmed that this is a bug, and I understand the cause. Now I just need to find the best fix.

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.84, which I just published. It will also be included in the next release of Pylance.
